lib/src/client/transport/cors.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides CORS support for HTTP based RPC requests.
+
+import '../query_parameter.dart';
+
+/// The default URL parameter name to overwrite http headers with a URL param
+/// to avoid CORS preflight.
+///
+/// This comes from the JS impl details at
+/// https://github.com/whatwg/fetch/issues/210#issue-129531743.
+const _httpHeadersParamName = '\$httpHeaders';
+
+// TODO(mightyvoice): Add the const parameter name for HTTP method if not
+// always use POST for gRPC-web.
+
+/// Manipulates the path and headers of the http request such that it will
```

Comment:
I would just keep the first paragraph of this comment and then shorten the rest (because it is irrelevant for understanding what function does and just makes reader overloaded with information), e.g. 

```dart
/// Manipulates the path and headers of the http request such that it will
/// not trigger a CORS preflight request and returns the new path with extra
/// query parameters.
///
/// Current implementation simply moves all headers into the special query
/// parameter which would be handled by a gRPC-web proxy.
///
/// For more info on exactly what may trigger CORS preflight, see
/// https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS.
```

Reply:
Done.